pkg/apis/authentication/types.go

```diff
@@ -88,3 +89,58 @@ type UserInfo struct {
 
 // ExtraValue masks the value so protobuf can generate
 type ExtraValue []string
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TokenRequest requests a token for a given service account.
+type TokenRequest struct {
+	metav1.TypeMeta
+	// ObjectMeta fulfills the metav1.ObjectMetaAccessor interface so that the stock
+	// REST handler paths work
+	metav1.ObjectMeta
+
+	Spec   TokenRequestSpec
+	Status TokenRequestStatus
+}
+
+// TokenRequestSpec contains client provided parameters of a token request.
+type TokenRequestSpec struct {
+	// Audiences are the intendend audiences of the token. A recipient of a
+	// token must identitfy themself with an identifier in the list of
+	// audiences of the token, and otherwise should reject the token. A
+	// token issued for multiple audiences may be used to authenticate
+	// against any of the audiences listed but implies a high degree of
+	// trust between the target audiences.
+	Audiences []string
+
+	// ValidityDuration is the requested duration of validity of the request. The
+	// token issuer may return a token with a different validity duration so a
+	// client needs to check the 'expiration' field in a response.
+	ValidityDuration int64
+
+	// BoundObjectRef is a reference to an object that the token will be bound to.
+	// The token will only be valid for as long as the bound objet exists.
+	BoundObjectRef *BoundObjectReference
```

Comment:
Feels like a last minute nit, but why not have this just be an owner ref?  Node access control?

Reply:
It's an interesting idea that hadn't occurred to me. Some problems i see with owners ref are we would like to enforce a single owner. Potentially thats the purpose of the Controller feel.

As documented:
> OwnerReference contains enough information to let you identify an owning
> object. Currently, an owning object must be in the same namespace, so there
> is no namespace field.

This seems vague enough that it might fit.

Reply:
Talking to @liggitt, this doesn't feel like a great fit for a few reasons:
* Owners references are intended to aid object garbage collection. The semantics are "if all objects in this list are deleted, then this object should be deleted". The semantics we would want for this API are "If any object in this list is deleted, the token should be invalid".
* We want to limit the number of bound objects, and right now we think we only want to support one. We would need to add specific validation to a generic field to support this.
I would prefer we don't overload owners references here.